component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -210,24 +207,32 @@ private void validateSubjectToken(OAuthTokenReqMessageContext tokReqMsgCtx, Map<
         impersonator = resolveImpersonator(claimsSet);
         if (StringUtils.isBlank(impersonator)) {
             // If claim values are empty, handle the exception
-            handleException(OAuth2ErrorCodes.INVALID_REQUEST, "Subject of actor token and impersonator are different.");
+            handleException(OAuth2ErrorCodes.INVALID_REQUEST, "Impersonator is not found in subject token.");
         }
+        String jwtIssuer = claimsSet.getIssuer();
```

Comment:
Don't we need to check if the jwtIssuer is null here?

Reply:
we are already checking it in validating mandatory claims section
https://github.com/wso2-extensions/identity-oauth2-grant-token-exchange/blob/954abdac72a814871f03904eeb825653f02f87df/component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java#L706

Reply:
ok